### ISSUE TITLE
Fix  'parseError' for custom variables in styledComponents

### ISSUE THREAD

**Issue by laurahw** [stylelint/stylelint#4583]
> Clearly describe the bug

I'm using a variable for screen breakpoints in StyledComponents elements and stylelint is catching some of them and throwing a 'parseError'. 

Example use: 
`  @media ${tabletBreakpoint} { ... }`

This is used in 66 files throughout the project, but is only being flagged on the most recent file to use it. 

> Which rule, if any, is the bug related to?

Not sure, terminal is not printing out any more detail than simply ` 24:3  ✖  Cannot parse selector   parseError`

> What CSS is needed to reproduce the bug?

StyledComponents: 

```
const Wrapper = styled.div`
  padding: 0 64px;

  @media ${tabletBreakpoint}{
    margin: 40px auto;
  }

`;
```

whereby `${tabletBreakpoint}` is equal to `'only screen and (min-width: 768px)'`

> What stylelint configuration is needed to reproduce the bug?

```
{
  "processors": [
    "stylelint-processor-styled-components"
  ],
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ],
  "rules": {
    "no-descending-specificity": null,
    "no-duplicate-selectors": true,
    "declaration-block-no-duplicate-properties": true,
    "declaration-block-no-redundant-longhand-properties": true,
    "color-no-hex": true
  }
}

```

> Which version of stylelint are you using?

`11.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `yarn run stylelint`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Related to variables in styledcomponents

> What did you expect to happen?

No warnings to be flagged. Or, alternatively, a problem to be flagged in the 65 other files where this is in place. 

> What actually happened (e.g. what warnings or errors did you get)?
✖  Cannot parse selector   parseError

This bug has been reported a number of times in various forms, but unfortunately whatever fixed other similar issues has not solved this. I'd really appreciate any ideas/help with this as have run into it a number of times.

**Comment by evilebottnawi** [comment]
Please do not use `stylelint-processor-styled-components`, it is deprecated, stylelint supports CSS-in-JS out of box

**Comment by laurahw** [comment]
Unfortunately, removing that from `.stylelintrc` doesn't fix the problem and brings up a whole lot of new ones. Do you have any suggestions for what I could replace it with?

**Comment by evilebottnawi** [comment]
Can you create simple reproducible repo?

**Comment by fanich37** [comment]
It's weird but I can't reproduce this case on the demo site :(
@laurahw could you please check your case with the latest `stylelint` version (13.0.0)?

**Comment by laurahw** [comment]
@fanich37  - good idea, and thanks for the suggestion, I hadn't tried that. Sadly, still the same error. 
It's now pointing to ` 24:3  ✖  Cannot parse selector   parseError`, which is just the line declaring the padding, but the error only crops up when we use custom media queries. 

It's really frustrating because I know that's why it's throwing errors, but it'll tell us to look at completely innocuous lines.

**Comment by jeddy3** [comment]
>> Please do not use stylelint-processor-styled-components, it is deprecated, stylelint supports CSS-in-JS out of box

> Unfortunately, removing that from .stylelintrc doesn't fix the problem and brings up a whole lot of new ones.

We're in the process of deprecating processors in favour of built-in syntaxes because processors aren't compatible with the autofix feature. However, the built-in syntaxes aren't up to scratch at the moment (see https://github.com/stylelint/stylelint/issues/4574), which has unfortunately made the migration a bit messy for everyone.

We are currently [improving our documentation](https://github.com/stylelint/stylelint/pull/4561) [around this](https://github.com/stylelint/stylelint/blob/855fa8a4b92e16e42d7a896b28e84537d9d87ff3/docs/about/syntaxes.md) to help improve the experience.

It would be beneficial for us if you could list out the new problems you encountered when removing the processor from your configuration file. It will help us build an accurate picture of what needs to be done to shore-up the built-in syntaxes.

> I know that's why it's throwing errors, but it'll tell us to look at completely innocuous lines.

Behind the scenes, stylelint transforms the JavaScript template literals (used by styled-components) into something that resembles CSS so that built-in rules understand it. This transformation can make it awkward to track down the origin of these types of errors.

It would be helpful if you could create a minimum reproducible example (either as a repo or using the [online demo](https://stylelint.io/demo) of your problem, i.e. try to recreate the problem with the minimum amount of code possible. It will help us resolve the issue.

**Comment by bartlangelaan** [comment]
Here is a demo with a few errors that showed up after we tried to ditch the custom processor in favour of the CSS-in-JS parser:

[Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96A3AB1OIkQHpxuAOoALHrmzyqAVxIlcKckgCGBIrx0APXEny40NAG5JYFG3r2WIaADSL5+GLkxqY5mEwaTHgJKS0oXRZKGB1sFUwYFkwdTgBzJJdAuKwWfDQdMCSAIxhKLG8ACgBRIzQYMGxyAiR0sgICotxS8sSfVIzcLJTsLABKMM1tHWjY+MTkgcz6kdz8wqSdSibMXBq6hqaKfFa09vWurZ3+9O9hnMwJ0U5ITkJcADE1EgBZaYBlAzYYwARlwAF4CMQyFAAHSyEEAAxEuFwAAEIrpcAASYBoCEAPgssMxOgAvrhgCjUbhICQsPAfORhJxUWSRIiWSIADxfdR-SI6QGGIxg0ngoQgSooVosADutiUjIArAAGVVoIwTEC4cQEsTPSZyBRKMymcxQLjeBVKbopMAAaxgfFwOhIiR0UCIFmstnI7lSrqgkWwSC4btwgWC5qZGBw5FCnEkU0FsziCSSnB0WhYnAgLBUnAdeblrMqAFVuAdGs1C8WIKWU1iYum+lmtLhJZKniJXu8+b8AUDjAAmCFQ5hwhHI1no0l7XH48FEtAk6ZksaU6moukMplQFlsjlczi874C3TC4FGMfiyUyzjyxWyFXqzWS3X6w3PZPG1lOHQcF0DR53wYcTDNZwbDsXAHFdH0XHcJRvHwbNvCjLAhjAMAEnMFAKhYXIyHwfBE2TUk03mTM0NzfM6xLMtKxgasjlweiG1ZecWyo3B228e9uwNPs8AABUA0M3QvIVwPHJgYXhJFqQxaY9gfJ8oCVHE8UJYkFQ02QNy3WcdwgekQn3Q9cHZThOQNbkxKAyShxFXA9KUcFgDVVUKT1A0DWTABJPAACsVHeLwYBQcwSCQJ1TDwD0oHMZDXUaFQIwAKR0KwhTATAkDQbB3FaQhYgoCBKFMNI83y9J4thT5MOY7M0DIdw5XkVlguynR8Dygq8DNMLml62C3RUO5KqcGg3A8M0MN2M1EjjI4yKkHKSAmlgnSIOUsCgNheqqWp6hrChJVXZCtFhYojoAYVMrB2R1UZun4kBLvka7bq8XdMGensXi4d4AHFnQesyPhoFAABUvu8SE5PIBSZxMszGUXHTPsimAbvux7-pZWznm5MHsAhrAodoOGcc-Pzv2TABBEh8AgXAYaIep-n6wr9Bcs1SlOCxAK8ChRrAEi1twWAwBIQC4jDR9bvSPJOhKMoKj2E7DmaE42hQtW3t6d6AEJBOeYTcDJinMCplAADl-COGmO0R6Fkenak-ox7Tl2JK7cZ+mATdhP6ySJk9SfBgm7cd0qoBd7w9RAVwQDAkUEHAEiWFlYL8BT8AuEoJA0kzqlZ0lXxiMlRly5pTsQDiZI-FogsiwYmvFF8GBXG3BvinpR1W8iwqiE77Bu974yG7+1vWg22wWHkIxx8nvvJXYLReGH9BiFXiap-ryUZblkZFZYAeICHvMWCgFRWqQMA4iSZwVlDfxO4AbT71EJ4Pn+jL1yPsgaqFQv4AOAX2BoKhQxWCSHfB+T8mj4CfEoW+SBKAxESNvDaE18BdmnvXAAugA6yxDD40mPg0U+CsuAX0Hg6Vu+BZBYCUKkA6r9AjEEInAoIfp8EIC7v-aekpyjb0oNmJAJAiCtwQTFJB1EtACMZH-Hu68QBiOwNESR0jW4yhIqcFgGRuD5TANomUuidp7UwFAfeaiREaMLI0c+T8SBmJvvPN0tg7EUNRKIpxoY6GuPcXRN4asDr3FGJgHx6jKABPPjFbggEjEpEiDwLRN88xvGBJwKAgEDqRESM4rgMSHE7WoDRE+8tAmPg8RwHAqRsClOARRbiGZcw0RvuxUszTKEgBvrAPqPBIgq3yA0DBj8kB70Eao3xDcBn33kc-FgTdUArWbtXGZa8HELMQcsrwZBGhYGUUI+xwCb4j24azBIRRel+P6fmZiE8ZheBlLuN4dz5n5k8TFApEAVADySPgOWzC2C0C3nwT5F0ZpcJkV09uHEoVpxgIcqJLA0BeBUJaNgIKUHwvrD0rZwjgEHMOLkDFMAsWPLIBC1u3TRBErOX00lRzMAsGIPUOlCLCUqO2SSiehjMkwDlIkmASLCxTK5QShlvLiV9KblXJIlyZGiovhrRI4DCENzdHKHQRABFzNRHXIB9zmJFEKpqk1wDL6OmIigq4gQ8idJojah0drJSGqAaIpAmBCC5iduQAhVrcBEM9fckuNUxWCO-lq4BDq2Wb3SUGq1JCtVkJpKG9RCqW58UOl4Tukp6RykCB6rNWjFUdJzBcTY2wS2CMlDqvVKC9ZnCSKK0tDjs1kDyJFR+j1anCtVfGgtjcSC6v1R261DCcUQBbRfe0Srd4qtaOrY2I7uC8MnX011M653FAXf6kVK6Vm1uifW0d46DXqJ3bLWdhj90bEPaqnoYDz2Nv1SwFAahQwsHbSna907b17oPdWtVa631jqbXkU43a-1zMlDurInB70Hu4Ee7gJ6dgjvfSgr9JAf1wYA1fRhSGUOPtA8OiDl7oP61-Surd9zEP1GQyrB9RRVaPpfRqqjTaGMzwJkvZieao2MkLQ2Ot8HC5mUE0YX9PA0hKBHcw1hfGN7gvSSwZVdGMNcZE7gGNJqG2QYnf+rVxrDMgDNTAC10aIF9OLr6jJAbbEgDs5m2NDcI2vsZAZ4NDckaJK0ZvdhAi3OkL7u5yB6nt4dSmf4NWOc3h+mw8Zq9DicKEFoOimFOAZFaefeqvTvm42pb40auzkorM2Z83Z+58a2BhVGCgbLwxplhvsz6v13B47JpNZFq1kovPcZqx5urp6wUoAhWVr1yAkuwBo627TJQGG9aAamk16bUT9fuVUs+dDlZpA4+xyjon+kwE3aZ4Bu3aFKwGEd1d3ntWlcu3067NT6HEZ7W8-tT7j0nae9RvDBH6MvZ29Q6p58d2vL7fSR8FHT0peoy22DIPJNvch9O6H7z7tgce5KDdEn1Ho-25jmDbbj0oGMLfcHe2PmCJBGjmnN2PtD2eVIwxWP+2I946DhuxPHzY7Q0OhHPGP1A6QEt1TIB+dgthzj-7Rmkdk8l7zqhssIdBJ+6B3T67zuE4cTL-Lx6de2a1YrnnnrzMzaq0003fntXjf59N4BDmuvOdW+QirIDI2Wr8w28bibeDO76SVP0C2Uc6ZW65jz62gGbZDZJuJLx3vBd+xhhXF6oPi5V4n+JmvJsvISxnnD4fyfcCl0n4pAvwWF844V3XF3c-J-PpT2TRvuud1VE3qvlakhP3zeeot+uXd58fAUbByEvB4sHce0PsBudi+-RLwjDjK-vfH+kzw-gcdz70-jvXbLkdl6jd397sXkFF5F6dnDUvWiwEMDUzuI5JNkHSGggAXoEfM8KplItb5ppdJbE5BndRVpOYdpbHeHLDUXNLFpaYSiCArXBLE3U7AnM9STMA1sRRPvI6EdIfdA0A+AtpBYDfXgLfZtBLXfBvYfPpTAqiRYW4QiFYB4eXK-AHC3Qg1MYgpIFIRgyJNYZA+vGAqXCiAARwmkwGXT9VTyFz+zYPN0X3w2X1R04KiHEMCCkKCxr1YOgOv1S1Lxz1UJmHUMkLo2kO0O1yENQIPylwuVMnlFkDi2rSRRvn0TnWGUIkoDyH+UwHY1kPL0ZUk04BUBQFKDZTIE9EMU-xoAX1gL6WCNCMdRvjZ0SUO2iNnSRU4Vy2EzwPEwIIcQrXbwe2G30wAQUNwyX0MLMy9xt192DW9Uc39R62j2DW2xmyGyK1qwdx2AmymxaJTXC2njaMlBZTRTiAFWKBgRKHtCdD4B3zmz3zO0b3UVGNyHGPykmKaCYOyDRSgLrWsOWIcVWLZXWKQE2OWB2IELr3AwOJoPuWOIm2KFaBYL2LPT0MvSlwePYCeKzF2MEJuPYJM0kweMGSKFyUaUeOeLRUyToWcMCJWJRTJXCLMC0ILzTxrV0MBLiPuMRNZTMNRMpxxxQPmRsNV2RVRVyH-yKJOS7wRIpLZQpSpRxV6hQX70WPwM+NxLRUZOxXV2zlIMn231AyoPPTQM5PpPRUxWxURNpTZNyOLXyJJS5PJSlOpRxm3kFwgGBEf3PUtABTIHFKRPZU5hwIH1Ow5LJIlSC1wMHzyKl1wSSCsX2hyNtIVPtPGnJ3MLRP8IxP2KxM-UqJX2AQdPxN6Nr2O3kMzw-SPyqODI9NDNT0sIBP30OLjM2k9K0SpMANFRpOpGsjJBADJCAA)

It looks like the JavaScript (whats inside of the template literals) is being parsed by StyleLint, instead of ignoring it. That's probabily the main difference with the 'old' processor, which I believes transformed all JS code to css variables to make it parsable by StyleLint.

**Comment by jeddy3** [comment]
@bartlangelaan Thank you for creating that demo. It highlights just how much work is left to do shoring up the postcss-css-in-js parser. We're [putting together a plan](https://github.com/stylelint/postcss-css-in-js/issues/3), and please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if you have time.

> EDIT: It seems like these issues were already reported in #4140.

I believe so. I will close this issue as we weren't able to reproduce the original problem using the demo. @laurahw Feel free to reopen the issue with a reproducible repo or demo, though.

**Comment by Dip686** [comment]
**It seems stylelint could not highlight the failing line properly, In my case, there was an issue but it was logging a different line in console.**
Debugging tips: try removing a few parts of the code and run stylelint to verify whether the issue is gone or not. In this way figure out the failing part. Divide and Conquer.

**Comment by eps1lon** [comment]
> Please do not use `stylelint-processor-styled-components`, it is deprecated, stylelint supports CSS-in-JS out of box

@alexander-akait I could not find a mention of this on [their GitHub repository](https://github.com/styled-components/stylelint-processor-styled-components) nor [their page on npmjs.com](https://www.npmjs.com/search?q=stylelint-processor-styled-components).

What makes you say that `stylelint-processor-styled-components` is deprecated?
